keras_cv/models/classification/image_classifier.py

```diff
@@ -0,0 +1,189 @@
+        super().__init__(**kwargs)
+        self.pooling = layers.GlobalAveragePooling2D(name="avg_pool")
+        self.dense = layers.Dense(
+            num_classes,
+            activation=activation,
+            name="predictions",
+        )
+        self.num_classes = num_classes
+        self.activation = activation
+
+    def call(self, inputs):
+        x = self.pooling(inputs)
+        return self.dense(x)
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "num_classes": self.num_classes,
+                "activation": self.activation,
+            }
+        )
+        return config
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class ImageClassifier(Task):
+    """Image classifier with configurable prediction head.
+
+    Args:
+        backbone: `keras_cv.models.Backbone`, the backbone architecture of the
+            classifier called on the inputs.
+        num_classes: int, number of classes to predict.
+        activation: A `str` or callable. The activation function to
+            use on the Dense layer. Set `classifier_activation=None` to return
+            the logits of the "top" layer.
+        head: keras.layers.Layer. Prediction head for classifier. Must take
+            arguments "num_classes" and "activation" and have no other
+            positional arguments.
+
+    Example:
+    ```python
+    input_data = tf.ones(shape=(8, 224, 224, 3))
+
+    # Pretrained backbone
+    model = keras_cv.models.ImageClassifier.from_preset(
+        "resnet50_v2_imagenet",
+        num_classes=4,
+    )
+    output = model(input_data)
+
+    # Randomly initialized backbone with a custom config
+    model = keras_cv.models.ImageClassifier(
+        backbone=keras_cv.models.ResNet50V2Backbone(),
+        num_classes=4,
+        head=keras_cv.models.classification.PooledDenseHead,
+    )
+    output = model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        backbone,
+        num_classes=2,
+        activation="softmax",
+        head=PooledDenseHead,
+        **kwargs,
+    ):
+        if isinstance(head, str):
+            if head == "pooled_dense":
+                head = PooledDenseHead
+            else:
+                ValueError(
+                    '`head` must be "pooled_dense" or a keras.layers.Layer. '
+                    f"Got: {head}"
+                )
+        head_fn = head(num_classes, activation)
+
+        inputs = backbone.input
+        x = backbone(inputs)
+        outputs = head_fn(x)
+
+        # Instantiate using Functional API Model constructor
+        super().__init__(
+            inputs=inputs,
+            outputs=outputs,
+            **kwargs,
+        )
+        # All references to `self` below this line
+        self.backbone = backbone
+        self.num_classes = num_classes
+        self.head = head
+        self.activation = activation
+
+        # Default compilation
```

Comment:
I like models "batteries included". It's nice to have the loss and metric built in.
However, I will change the learning rate. How do I do that while leaving the default loss and metrics ?

Reply:
There are actually 2 possible default cases, btw:

1. 2-class classification -> should use `num_classes=1, activation="sigmoid"` and `loss="binary_crossentropy"`
2. More than 2 classes (current settings)

How should we handle that? Should we pick the right defaults based on `num_classes`? (like we do with "accuracy")

Reply:
I am pretty anti "batteries-included".  This is a meaningful choice users should make based on their dataset.

Do you only have a few hard examples in your entire dataset (class imbalance?).  You need focal loss.  sure you can make a somewhat sensible default, but I think batteries included is an anti-pattern personally.